core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -210,6 +211,20 @@ private TableProperties() {
   public static final String SPARK_WRITE_PARTITIONED_FANOUT_ENABLED = "write.spark.fanout.enabled";
   public static final boolean SPARK_WRITE_PARTITIONED_FANOUT_ENABLED_DEFAULT = false;
 
+  public static final String PARTITIONED_FANOUT_WRITERS_CACHE_SIZE = "write.partition.fanout.writers-cache-size";
```

Comment:
I am not sure if the configure should be table properties. If the table has a spark and a flink writer, the size is hard to configure.